crypto/fipsmodule/evp/evp.c

```diff
@@ -290,8 +290,12 @@ EVP_PKEY *EVP_PKEY_new_mac_key(int type, ENGINE *engine, const uint8_t *mac_key,
   if(key == NULL) {
     goto err;
   }
-  key->key = mac_key;
+  key->key = OPENSSL_memdup(mac_key, mac_key_len);
+  if (key->key == NULL && mac_key_len > 0) {
+    return 0;
```

Comment:
will this leak memory after the allocation of `ret`? other error cases `goto err;`

Reply:
Ahhh nice catch